src/components/EditorWidgets/Markdown/MarkdownControl/Toolbar/Toolbar.js

```diff
@@ -27,9 +27,75 @@ export default class Toolbar extends React.Component {
     };
   }
 
-  isHidden = button => {
+  isHidden = (button, isDefaultButton = true) => {
     const { buttons } = this.props;
-    return List.isList(buttons) ? !buttons.includes(button) : false;
+    return List.isList(buttons) ? !buttons.includes(button) : !isDefaultButton;
+  }
+
+  shouldDisplayHeadingsList() {
+    const { buttons } = this.props;
+    return List.isList(buttons) && buttons.includes('headings-list');
+  }
+
+  getHeadingsList() {
+    const {
+      onBlockClick,
+      selectionHasBlock,
+      disabled
+    } = this.props;
+
+    return (
+      <div className="nc-toolbar-dropdown">
+        <Dropdown
+          dropdownTopOverlap="36px"
+          button={
+            <ToolbarButton
+              label="Headings"
+              icon="text"
+              onClick={this.handleComponentsMenuToggle}
+              disabled={disabled}
+            />
+          }
+        >
+          <ToolbarButton
+            type="heading-one"
+            label="Header 1"
```

Comment:
I think these should be `Heading` not `Header` as a `<header>` is a very different element and this could cause confusion.